src/test/java/io/xpring/xrpl/ProtocolBufferConversionTest.java

```diff
@@ -289,4 +289,90 @@ public void convertSignerWithAllFieldsSetTest() {
         assertThat(xrpSigner.transactionSignature())
                 .isEqualTo(signerProto.getTransactionSignature().getValue().toByteArray());
     }
+
+    // Transaction
+
+    @Test
+    public void convertPaymentTransactionWithAllCommonFieldsSetTest() {
+        // GIVEN a Transaction protocol buffer with all common fields set.
+        Transaction transactionProto = FakeXRPProtobufs.transactionWithAllFieldsSet;
+
+        // WHEN the protocol buffer is converted to a native Java type.
+        XRPTransaction xrpTransaction = XRPTransaction.from(transactionProto);
+
+        // THEN all fields are present and converted correctly.
+        assertThat(xrpTransaction.account()).isEqualTo(transactionProto.getAccount().getValue().getAddress());
+        assertThat(xrpTransaction.accountTransactionID())
+                .isEqualTo(transactionProto.getAccountTransactionId().getValue().toByteArray());
+        assertThat(xrpTransaction.fee()).isEqualTo(transactionProto.getFee().getDrops());
+        assertThat(xrpTransaction.flags()).isEqualTo(transactionProto.getFlags().getValue());
+        assertThat(xrpTransaction.lastLedgerSequence()).isEqualTo(transactionProto.getLastLedgerSequence().getValue());
+        assertThat(xrpTransaction.memos()).isEqualTo(transactionProto.getMemosList()
+                .stream()
+                .map(memo -> XRPMemo.from(memo))
+                .collect(Collectors.toList()));
+        assertThat(xrpTransaction.sequence()).isEqualTo(transactionProto.getSequence().getValue());
+        assertThat(xrpTransaction.signers()).isEqualTo(transactionProto.getSignersList()
+                .stream()
+                .map(signer -> XRPSigner.from(signer))
+                .collect(Collectors.toList()));
+        assertThat(xrpTransaction.signingPublicKey())
+                .isEqualTo(transactionProto.getSigningPublicKey().getValue().toByteArray());
+        assertThat(xrpTransaction.sourceTag()).isEqualTo(transactionProto.getSourceTag().getValue());
+        assertThat(xrpTransaction.transactionSignature())
+                .isEqualTo(transactionProto.getTransactionSignature().getValue().toByteArray());
+        assertThat(xrpTransaction.type()).isEqualTo(TransactionType.PAYMENT);
+        assertThat(xrpTransaction.paymentFields()).isEqualTo(XRPPayment.from(transactionProto.getPayment()));
```

Comment:
This seems a little odd to me. This looks like you're basically asserting that you wrote this same logic in the implementation, like:
```java
        assertThat(xrpTransaction.memos()).isEqualTo(transactionProto.getMemosList()
                .stream()
                .map(memo -> XRPMemo.from(memo))
                .collect(Collectors.toList()));
```

Is it too hard to have a set of hardcoded data/values to compare against?

Reply:
This is a good idea - I made myself a TODO among quite a few other SDK cleanups that will be happening after getting these initial wrapper classes / paymentHistory merged.  I think it would make sense to make another test object that can live in the FakeXRPProtobufs alongside the object whose conversion it's representing.